corehq/apps/users/bulk_download.py

```diff
@@ -43,84 +47,136 @@ def build_data_headers(keys, header_prefix='data'):
+    """
+    Returns a comma-separated list of IMEI numbers of the user's devices, sorted with most-recently-used first
+    """
+    return ', '.join([device.device_id for device in sorted(
+        user.devices, key=lambda d: d.last_used, reverse=True
+    )])
+
+
+def make_mobile_user_dict(user, group_names, location_cache, domain, fields_definition):
+    model_data = {}
+    uncategorized_data = {}
+    model_data, uncategorized_data = (
+        fields_definition.get_model_and_uncategorized(user.metadata)
+    )
+    role = user.get_role(domain)
+    profile = None
+    if PROFILE_SLUG in user.metadata and domain_has_privilege(domain, privileges.APP_USER_PROFILES):
+        try:
+            profile = CustomDataFieldsProfile.objects.get(id=user.metadata[PROFILE_SLUG])
+        except CustomDataFieldsProfile.DoesNotExist:
+            profile = None
+    activity = user.reporting_metadata
+
+    location_codes = []
+    try:
+        location_codes.append(location_cache.get(user.location_id))
+    except SQLLocation.DoesNotExist:
+        pass
+    for location_id in user.assigned_location_ids:
+        # skip if primary location_id, as it is already added to the start of list above
+        if location_id != user.location_id:
+            try:
+                location_codes.append(location_cache.get(location_id))
+            except SQLLocation.DoesNotExist:
+                pass
+
+    def _format_date(date):
+        return date.strftime('%Y-%m-%d %H:%M:%S') if date else ''
+
+    return {
+        'data': model_data,
+        'uncategorized_data': uncategorized_data,
+        'group': group_names,
+        'name': user.full_name,
+        'password': "********",  # dummy display string for passwords
+        'phone-number': user.phone_number,
+        'email': user.email,
+        'username': user.raw_username,
+        'language': user.language,
+        'user_id': user._id,
+        'is_active': str(user.is_active),
+        'User IMEIs (read only)': get_devices(user),
+        'location_code': location_codes,
+        'role': role.name if role else '',
+        'user_profile': profile.name if profile else '',
+        'registered_on (read only)': _format_date(user.created_on),
+        'last_submission (read only)': _format_date(activity.last_submission_for_user.submission_date),
+        'last_sync (read only)': activity.last_sync_for_user.sync_date,
+    }
+
+
+def make_web_user_dict(user, domain):
+    user = CouchUser.wrap_correctly(user['doc'])
+    domain_membership = user.get_domain_membership(domain)
+    role = UserRole.get(domain_membership.role_id).name
+    return {
+        'first_name': user.first_name,
+        'last_name': user.last_name,
+        'email': user.email,
+        'username': user.username,
+        'status': 'Active User',
+        'role': role,
+        'last_access_date (read only)': domain_membership.last_accessed,
+        'last_login (read only)': user.last_login,
+        'remove': '',
```

Comment:
I checked and mobile worker download does not have the removal column, but the "is_active" column can be changed to False for deactivation of mobile workers. I couldn't find docs about the 'remove_web_user' column for the mobile upload to remove the membership of the web user (could be there and I just didn't find it first pare through). My thought for adding the remove column now was that then the remove would be there for the upload, but I see how that causes confusion, and also download is used for more than just the then immediately upload

Reply:
That all makes sense - I think it could go either way. If people see "remove" in the download they may get confused, but if they have to add "remove" on upload, they'll forget about it and it's going to get misspelled. Might be something to mention to product (Kat/Dev).

Reply:
I think "is_active" is a little different because it's a property, and because `remove_web_user` is a USH thing I don't think that decision got product attention.

Reply:
Ahh I see-- yes I will see what they say

Reply:
agree that this seems confusing

Reply:
@orangejenny @calellowitz spoke with Kat regarding the 'remove' column in the download form. She's leaning toward it being included in download just for the sake of consistency between upload and download forms. Since there is some level of confusion either way, it would be good to discuss further or keep an eye on and get feedback once released